resolver/types.go

```diff
@@ -164,8 +164,52 @@ type ValidationRule struct {
 }
 
 type ValidationError struct {
-	Code code.Code
-	Rule *CELValue
+	Code    code.Code
+	Rule    *CELValue
+	Details MessageValidationDetails
+}
+
+// ReferenceNames returns all the unique reference names in the error definition.
+func (v *ValidationError) ReferenceNames() []string {
```

Comment:
Please move this logic to resolver/validation.go, since I want to keep resolver/types.go to type definitions and minimal implementation.